### DISCUSSION TITLE
Friendly ideas for mv-action attributes

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#884] [Ideas]
Hi

Based on #883, wouldn't it be nice to extend mv-action attributes with following some 'natural' ones like:

- duplicate(), to duplicate a item inside same collection
- copy(), to duplicate an item with its values into another collection
- update(), to update  a property value inside target collection regarding it's new value in source collection
- deleteSilent() , to delete wihtout a toast message
- [deleteWithMessage](https://github.com/mavoweb/mavo/discussions/883#discussioncomment-3116012)(), to delete with a custom message toast 

most of them can use existing mv-action attributes, but I found them more user friendly

not related, but:
Do mv-action attributes continue to execute, if you deactivate expressions?

**Comment by karger** [comment]
I wasn't proposing to *rename* delete as deleteWithMessage, because I think in many cases people still want the default message.  I would like to have a different action for specifying my own message though.

**Reply by GalinhaLX** [reply]
sorry for missunderstood, I corrected initial message

**Comment by LeaVerou** [comment]
Hi @GalinhaLX,

Duplicate and copy are already easily possible:
- For duplicate, you just use `add(item)`, this is even in the [stock example](https://mavo.io/docs/actions#add)
- For copy, you do the same as duplicate, but specify a different collection

Not sure I understand update().

> Do mv-action attributes continue to execute, if you deactivate expressions?

You mean through `mv-expressions="none"`? Yes, I believe so. Even `mv-value` would still work.

**Reply by GalinhaLX** [reply]
Hi @LeaVerou 

What I mean with 'update':

Let's say you have 2 collections 
(in different apps, both with different storages), and you want to 'backup' collection1 into collection2
-> What I've done till now, is that I clear collection2 , then add collection1 into collection2 (works)

BUT sometimes you have critical information on collection2, and you can't have Collection2 empty, 
so what happens in case communication is down just after delete, without the add action : collection2 will be empty :(

So the idea to always have something in Collection2, would be to update only modified items from Collection1 into Collection2

Don't know if Mavo can handle this case (or if it already does), but maybe using  'contains' function in a loop

**Reply by LeaVerou** [reply]
What about `set(collection2, collection1)`?

**Comment by GalinhaLX** [comment]
I created this [codepen](https://codepen.io/galinhalx/pen/GRxmNNQ) to resume answers

only delete_silent() missing for now
